src/main/java/org/spongepowered/api/command/parameter/managed/standard/CatalogedValueParameters.java

```diff
@@ -0,0 +1,432 @@
+     * <p>Returns a {@link Location}.</p>
+     */
+    public static final CatalogedValueParameter<Location> LOCATION =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "LOCATION");
+
+    /**
+     * Require an argument to be a long (base 10).
+     *
+     * <p>Returns a {@link Long}.</p>
+     */
+    public static final CatalogedValueParameter<Long> LONG =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "LONG");
+
+    /**
+     * Does not parse any arguments, returning nothing.
+     *
+     * <p>Returns nothing - no entry will be placed into any provided key.</p>
+     */
+    public static final CatalogedValueParameter<Void> NONE =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "NONE");
+
+    /**
+     * Expect an argument to represent an online player.
+     *
+     * <p>This parameter accepts selectors.</p>
+     *
+     * <p>Returns a {@link Player}.</p>
+     */
+    public static final CatalogedValueParameter<Player> PLAYER =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "PLAYER");
+
+    /**
+     * Expect an argument to represent a {@link PluginContainer}'s id.
+     *
+     * <p>Returns a {@link PluginContainer}</p>
+     */
+    public static final CatalogedValueParameter<PluginContainer> PLUGIN =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "PLUGIN");
+
+    /**
+     * Require one or more strings, without any processing, which are combined
+     * into a single, space-separated string.
+     *
+     * <p>Returns a {@link String}.</p>
+     */
+    public static final CatalogedValueParameter<String> REMAINING_JOINED_STRINGS =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "REMAINING_JOINED_STRINGS");
+
+    /**
+     * Require one or more strings, which are combined into a single,
+     * space-separated string.
+     *
+     * <p>Returns a {@link String}.</p>
+     */
+    public static final CatalogedValueParameter<String> REMAINING_RAW_JOINED_STRINGS =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "REMAINING_RAW_JOINED_STRINGS");
+
+    /**
+     * Require an argument to be a string.
+     *
+     * <p>Returns a {@link String}.</p>
+     */
+    public static final CatalogedValueParameter<String> STRING =
+            DummyObjectProvider.createExtendedFor(CatalogedValueParameter.class, "STRING");
+
+    /**
+     * Does not parse any arguments, but instead returns a
+     * {@link org.spongepowered.api.block.BlockSnapshot} if the
+     * current {@link CommandSource} (which thus
+     * must be a {@link org.spongepowered.api.world.Locatable}).
+     *
+     * <p>This will always fail for non-locatable sources</p>
+     *
+     * <p>Returns a {@link org.spongepowered.api.block.BlockSnapshot}.</p>
+     */
+    public static final CatalogedValueParameter<BlockSnapshot> TARGET_BLOCK =
```

Comment:
There is no parameter to specify a block simply via coordinates (`CatalogedValueParameters.BLOCK`). There are no corresponding static methods in Parameter class for `BLOCK` and `TARGET_BLOCK`.

Reply:
> There is no parameter to specify a block simply via coordinates (CatalogedValueParameters.BLOCK).

Yes there is, `LOCATION`. A `Location` will be specified by co-ordiantes [and can get that information for you](https://github.com/SpongePowered/SpongeAPI/blob/d6a82f27cd42ab2a0fd7b8f06cb78475187cbc5a/src/main/java/org/spongepowered/api/world/Location.java#L225). 

> There are no corresponding static methods in Parameter class for BLOCK and TARGET_BLOCK.

`TARGET_BLOCK` doesn't need to be the `Parameter` class - not every parameter type needs to be represented there - it's intended to be easy access and we could remove some of the helper methods yet, it does feel a bit unwieldly.

Reply:
Okay... The fact is that I need an argument that would support ray tracing in order to detect the block that the player is looking at. (https://github.com/SpongePowered/SpongeAPI/issues/1917, https://github.com/SpongePowered/SpongeAPI/issues/1935)